src/main/java/core/basesyntax/service/operations/SupplyOperation.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.exceptions.InvalidFruitException;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (Storage.containsKey(transaction.getFruit())) {

```

Comment:
In the case of a supply operation, if the fruit is not already in the storage, it should be added instead of throwing an exception. Consider changing the behavior to add the fruit to the storage if it does not exist.

Reply:
ok, done